lib/events.js

```diff
@@ -332,12 +364,11 @@ EventEmitter.prototype.removeAllListeners =
 
       if (typeof listeners === 'function') {
         this.removeListener(type, listeners);
-      } else if (Array.isArray(listeners)) {
+      } else if (listeners && listeners.length) {
         // LIFO order
-        while (listeners.length)
+        while (listeners[0])
```

Comment:
Suggestion: turn this into a do/while loop.  It's already been verified that the list is not empty at this point.